tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/service/websocket/FailoverClientWithRetry.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http:www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http:www.apache.orglicensesLICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specif ic language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.test.service.websocket;
+
+import org.ballerinalang.test.context.BallerinaTestException;
+import org.ballerinalang.test.util.websocket.client.WebSocketTestClient;
+import org.ballerinalang.test.util.websocket.server.WebSocketRemoteServer;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.net.URISyntaxException;
+import java.nio.ByteBuffer;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This Class tests support retry function for failover WebSocket client.
+ */
+@Test(groups = {"websocket-test"})
+public class FailoverClientWithRetry extends WebSocketTestCommons {
+
+    private WebSocketRemoteServer remoteServer15100;
+    private String url = "ws://localhost:30004";
+    private int port = 15100;
+    private WebSocketRemoteServer remoteServer15200 = new WebSocketRemoteServer(15200);
+
+    @Test(description = "Tests sending and receiving text frames for retry function of the WebSocket failover client")
```

Comment:
Shall we add a exact description of what you are doing as the test is very long to be self explanatory?